anitab-org/bridge-in-tech-backend
# Feat: PUT and GET organization and tests

### Description
Create PUT and GET /organization API endpoints

Fixes #105 

### Type of Change:
- Code

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
* Run and test the application manually:
- Login as user who already has additional_info in the database and `is_organization_rep` is True
- try to send PUT /organization request before setting Cookie `user_id` through GET /organization. see error 403 Forbidden and error message
<img width="1397" alt="Screen Shot 2020-08-11 at 9 02 35 pm" src="https://user-images.githubusercontent.com/29667122/89976213-2666a180-dcab-11ea-99e7-0718c46b1b79.png">

- send GET /organization request. Since this user has not created an organization, see error 404 Not Found and message `Organization does not exist`.
<img width="843" alt="Screen Shot 2020-08-11 at 1 08 11 am" src="https://user-images.githubusercontent.com/29667122/89976026-b22bfe00-dcaa-11ea-8056-cc079bf5ca36.png">

- now create organization by sending PUT /organization request with correct payload. see code 201 Created and success message
<img width="1402" alt="Screen Shot 2020-08-11 at 9 30 29 pm" src="https://user-images.githubusercontent.com/29667122/89976257-46966080-dcab-11ea-95fc-6fc4bd4da686.png">

- now try to update the organization by changing some details. On correct update, see code 200 and success message
<img width="1407" alt="Screen Shot 2020-08-11 at 9 34 01 pm" src="https://user-images.githubusercontent.com/29667122/89976310-70e81e00-dcab-11ea-91b1-41ed528db0c2.png">

- logout and login as a user that either has additional_info `is_organization_rep` is `False` or has not created an additional_info. 
- send GET /organization request. see error 403 Forbidden and error message user has not declared `is_organization_rep` value
<img width="1393" alt="Screen Shot 2020-08-11 at 1 16 48 am" src="https://user-images.githubusercontent.com/29667122/89976496-e6ec8500-dcab-11ea-9677-f1262b11d97b.png">

- an example of successful GET /organization request will be as below
<img width="1407" alt="Screen Shot 2020-08-12 at 11 38 28 pm" src="https://user-images.githubusercontent.com/29667122/90021988-491da800-dcf5-11ea-82d8-11528ba8b581.png">

* Run unittests and ensure all tests passed


### Checklist:


**Code/Quality Assurance Only**

**Additional Info**
This is still a work in progress
